tests/modules/images/webp-uploads/load-tests.php

```diff
@@ -918,11 +920,63 @@ public function it_should_create_mime_types_for_allowed_sizes_only_via_global_va
 		$this->assertImageNotHasSizeSource( $attachment_id, 'not_allowed_size_200x150', 'image/webp' );
 	}
 
+	/**
+	 * Test image quality for image conversion.
+	 *
+	 * @test
+	 */
+	public function test_set_quality_with_image_conversion() {
+		global $wp_version;
+		// Set conversions for uploaded images.
+		add_filter( 'image_editor_output_format', array( $this, 'image_editor_output_formats' ) );
+		$expected_webp = 82;
+		if ( version_compare( '6.2', $wp_version ) ) {
+			$expected_webp = 86;
+		}
+
+		$editor = wp_get_image_editor( TESTS_PLUGIN_DIR . '/tests/testdata/modules/images/dice.png', array( 'mime_type' => 'image/png' ) );
+
+		// Quality setting for the source image. For PNG the fallback default of 82 is used.
+		$this->assertSame( 82, $editor->get_quality(), 'Default quality setting for PNG is 82.' );
+
+		// A PNG image will be converted to WEBP whose quality should be 82 for wp version 6.2 and above.
+		$editor->save();
+		$this->assertSame( $expected_webp, $editor->get_quality(), 'Output image format is WEBP. Quality setting for it should be 82 for WP version 6.2 and above.' );
+
+		unset( $editor );
+
+		$editor = wp_get_image_editor( TESTS_PLUGIN_DIR . '/tests/testdata/modules/images/leafs.jpg' );
+
+		// Quality setting for the source image. For JPG the fallback default of 82 is used.
+		$this->assertSame( 82, $editor->get_quality(), 'Default quality setting for JPG is 82.' );
+
+		// A JPG image will be converted to WEBP whose quality should be 82 for wp version 6.2 and above.
+		$editor->save();
+		$this->assertSame( $expected_webp, $editor->get_quality(), 'Output image format is WEBP. Quality setting for it should be 82 for WP version 6.2 and above.' );
+
+		unset( $editor );
+		remove_filter( 'image_editor_output_format', array( $this, 'image_editor_output_formats' ) );
+	}
```

Comment:
The logic in this test also needs to be updated based on my above comments, as it's currently not testing the right thing.

Reply:
Updated the logic to reflect the accurate changes in the test case as well.